src/ui/datacore-view-page.tsx

```diff
@@ -0,0 +1,241 @@
+import { debounce, ItemView, Menu, ViewStateResult, WorkspaceLeaf } from "obsidian";
+import { ScriptLanguage } from "utils/javascript";
+import { DatacoreJSRenderer, ReactRenderer } from "./javascript";
+import { DatacoreLocalApi } from "api/local-api";
+import { DatacoreApi } from "api/api";
+import { createContext } from "preact";
+import { Group, Stack } from "api/ui/layout";
+import { useCallback, useContext, useEffect, useState } from "preact/hooks";
+import { Textbox, VanillaSelect } from "api/ui/basics";
+import { useIndexUpdates, useStableCallback } from "./hooks";
+import DatacorePlugin from "main";
+import { DATACORE_CONTEXT } from "./markdown";
+
+export const VIEW_TYPE_DATACORE = "datacore-query-view";
+
+const CUSTOM_VIEW_CONTEXT = createContext<DatacoreQueryView>(undefined!);
+
+const backButton = (
+    <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill="none"
+        stroke="currentColor"
+        stroke-width="2"
+        stroke-linecap="round"
+        stroke-linejoin="round"
+        class="svg-icon lucide-arrow-left"
+    >
+        <path d="m12 19-7-7 7-7"></path>
+        <path d="M19 12H5"></path>
+    </svg>
+);
+
+function useVaultFiles() {
+    const core = useContext(DATACORE_CONTEXT);

```

Comment:
This is going to be a very slow hook in larger vaults since you are mapping over potentially tens of thousands of files on every single index update. The "right" way to do this is to change the thing using the file listing (the VanillaSelect below) to fetch the latest list of files whenever you open the dropdown. This might require using an actual fancy react dropdown instead of a vanilla `<select>` field? Maybe you can make it a button that when you click it becomes a dropdown?